setup.py

```diff
@@ -0,0 +1,127 @@
+from setuptools import setup, Extension, distutils, Command, find_packages
+import setuptools.command.install
+import setuptools.command.develop
+import setuptools.command.build_py
+import distutils.command.build
+from shutil import copyfile
+import subprocess, os
+
+TOP_DIR = os.path.realpath(os.path.dirname(__file__))
+
+print('TOP_DIR: ', TOP_DIR)
+
+###############################################################################
+# Version and build number
+###############################################################################
+
+try:
+    git_version = subprocess.check_output(
+        ['git', 'rev-parse', 'HEAD'], cwd=TOP_DIR
+    ).decode('ascii').strip()
+except (OSError, subprocess.CalledProcessError):
+    git_version = None
+
+tc_version = '0.1.0'
+tc_build_number = 2
+# versioning should comply with
+# https://www.python.org/dev/peps/pep-0440/#public-version-identifiers
+# when conda packaging, we get these values from environment variables
+if os.getenv('TC_BUILD_VERSION') and os.getenv('TC_BUILD_NUMBER'):
+    assert os.getenv('TC_BUILD_NUMBER') is not None, "Please specify valid build number"
+    tc_build_number = int(os.getenv('TC_BUILD_NUMBER'))
+    tc_version = str(os.getenv('TC_BUILD_VERSION'))
+    if tc_build_number > 1:
+        tc_version += '.post' + str(tc_build_number)
+
+print('git_version: {} tc_version: {} tc_build_number: {}'.format(
+    git_version, tc_version, tc_build_number
+))
+
+################################################################################
+# Copy the proto files only
+################################################################################
+copyfile("src/proto/compcache.proto", "tensor_comprehensions/compilation_cache.proto")
```

Comment:
Could you please explain why this and the ```.gitignore``` change above are needed versus say just using the particular file?
If it is because we don't want to ship stuff from ```src``` then maybe it is appropriate to have a ```proto/*.proto``` outside of src and use that without copying? 
What do you think @ftynse @ttheodor ?

Reply:
as part of the conda package, we don't ship the `src/proto` codepath or any of `src/`

hence, we just pick whatever is relevant and we only need the proto files.

Usage:

look at `tc_unit/decode` function. This is a functionality people wanted that once they have autotuned their options, the want to see what they look like. so now they can do

`tc.decode(cache_file)`

Reason for chosing name `decode` is because `protoc --decode` is the actual command and it was intuitive to use this name and user mentioned also "how do I decode the binary options file"?

Reply:
yeah, regarding `proto/` we only ship them in conda and not look at `src/`

Reply:
I'll mark this as resolved, current shipping of proto seems fine and the decode naming is okay too